src/contracts/pods/EigenPodManager.sol

```diff
@@ -651,29 +634,26 @@ contract EigenPodManager is
     }
 
     /**
-     * @notice Internal function to enter `podOwner` into undelegation limbo
+     * @notice Internal function to enter `podOwner` into undelegation limbo, and return the number of shares which were moved into undelegation limbo.
      * @dev Does nothing if the `podOwner` has no delegated shares (i.e. they are already in undelegation limbo or have no shares)
      * OR if they are not actively delegated to any operator.
+     *
+     * This method assumes the podOwner is delegated, as it's being called by the
+     * DelegationManager (and supplied with `delegatedTo`)
      */
-    function _enterUndelegationLimbo(address podOwner) internal {
-        if (!podOwnerHasNoDelegatedShares(podOwner) && delegationManager.isDelegated(podOwner)) {
-            // look up the address that the pod owner is currrently delegated to in EigenLayer
-            address delegatedAddress = delegationManager.delegatedTo(podOwner);
-
+    function _enterUndelegationLimbo(address podOwner, address delegatedTo) internal returns (uint) {
+        if (podOwnerShares[podOwner] != 0 && !isInUndelegationLimbo(podOwner)) {
             // store the undelegation limbo details
             _podOwnerUndelegationLimboStatus[podOwner].active = true;
             _podOwnerUndelegationLimboStatus[podOwner].startBlock = uint32(block.number);
-            _podOwnerUndelegationLimboStatus[podOwner].delegatedAddress = delegatedAddress;
+            _podOwnerUndelegationLimboStatus[podOwner].delegatedAddress = delegatedTo;
 
             // emit event
             emit UndelegationLimboEntered(podOwner);
 
-            // undelegate all shares
-            uint256[] memory shareAmounts = new uint256[](1);
-            shareAmounts[0] = podOwnerShares[podOwner];
-            IStrategy[] memory strategies = new IStrategy[](1);
-            strategies[0] = beaconChainETHStrategy;
-            delegationManager.decreaseDelegatedShares(podOwner, strategies, shareAmounts, true/*undelegateIfPossible*/);
+            return podOwnerShares[podOwner];
```

Comment:
would prefer to simplify the conditional here to `if (!isInUndelegationLimbo(podOwner))` and `return podOwnerShares[podOwner];` outside of the conditional

Reply:
hmm, that would be a meaningful change though - because now users that have never touched eigenpods would get state changes via this method.

Reply:
i.e. someone trying to undelegate that purely holds LSTs in stmgr would not be able to skip the conditional in your version

Reply:
what i think we want to capture with the conditional is a clear semantic distinction between addresses that have opted into restaking, and addresses that have not.

maybe we leave this as is and flag this as a future change?

Reply:
not sure I'm fully understanding -- we agreed we want the property that someone can't enter undelegation limbo if they have zero shares, right? can you write the suggestion you have?

Reply:
Yep, I was ok with that, but don't feel strongly